src/Database/LSMTree/Internal.hs

```diff
@@ -1278,3 +1306,140 @@ duplicate t@Table{..} = do
             tableConfig
             tableArenaManager
             content
+
+
+{-------------------------------------------------------------------------------
+   Table union
+-------------------------------------------------------------------------------}
+
+{-# SPECIALISE union :: Table IO h -> Table IO h -> IO (Table IO h) #-}
+-- | See 'Database.LSMTree.Normal.union'.
+union ::
+     (MonadMask m, MonadMVar m, MonadST m, MonadSTM m)
+  => Table m h
+  -> Table m h
+  -> m (Table m h)
+union t1 t2 = unions $ V.fromList [t1, t2]
+
+{-# SPECIALISE unions :: V.Vector (Table IO h) -> IO (Table IO h) #-}
+-- | See 'Database.LSMTree.Normal.unions'.
+unions ::
+     (MonadMask m, MonadMVar m, MonadST m, MonadSTM m)
+  => V.Vector (Table m h)
```

Comment:
Would be nice to use something non-empty here to avoid all the emptiness checks further down. I think a non-empty list would be okay. The overhead should be insignificant.